docs/docs/code/go/README.md

```diff
@@ -1,6 +1,452 @@
+
+You can also send files within a step.
+
+An example of sending a previously stored file in the workflow's `/tmp` directory: 
+
+```go
+package main
+
+import (
+  "os"
+  "log"
+  "mime/multipart"
+  "bytes"
+  "io"
+  "mime/multipart"
+  "net/http"
+  "os"
+)
+
+func main() {
+  // Instantiate a new HTTP client, body and form writer
+  client := http.Client{}
+  body := &bytes.Buffer{}
+  writer := multipart.NewWriter(body)
+
+  // Create an empty file to start
+  fw, err = writer.CreateFormFile("file", "go-logo.svg")
+  if err != nil {
+    log.Fatalln(e)
+  }
+
+  // Retrieve a previously saved file from workflow storage
+  file, err := os.Open("/tmp/go-logo.svg")
+  if err != nil {
+    log.Fatalln(e)
+  }
+
+  // Close multipart form writer
+  writer.Close()
+
+  _, err = io.Copy(fw, file)
+
+  // Send the POST request
+  req, err := http.NewRequest("POST", "https://postman-echo.com/post", bytes.NewReader(body.Bytes()))
+  if e != nil {
+    log.Fatalln(e)
+  }
+
+  req.Header.Set("Content-Type", writer.FormDataContentType())
+  rsp, err := client.Do(req)
+  if e != nil {
+    log.Fatalln(e)
+  }
+}
+```
+
+## Sharing data between steps
+
+A step can accept data from other steps in the same workflow, or pass data downstream to others.
+
+This makes your steps even more powerful, you can compose new workflows and reuse steps.
+
+### Using data from another step
+
+In Go steps, data from the initial workflow trigger and other steps are available in the `pipedream-go` package.
+
+In this example, we'll pretend this data is coming into our HTTP trigger via POST request.
+
+```json
+{
+  "id": 1,
+  "name": "Bulbasaur",
+  "type": "plant"
+}
+```
+
+In our Go step, we can access this data in the `Steps` variable from the `pd` package. Specifically, this data from the POST request into our workflow is available in the `trigger` dictionary item. 
+
+```go
+package main
+
+import (
+	"fmt"
+	"github.com/PipedreamHQ/pipedream-go"
+)
+
+func main() {
+	// Access previous step data using pd.Steps
+	fmt.Println(pd.Steps)
+}
+```
+
+### Sending data downstream to other steps
+
+To share data created, retrieved, transformed or manipulated by a step to others downstream call the `Export` function from `pd` package.
+
+An example speaks a thousand words, so here's one passing data from an API to the bash step.
+
+```go
+package main
+
+import (
+	"encoding/json"
+	"github.com/PipedreamHQ/pipedream-go"
+	"io/ioutil"
+	"net/http"
+)
+
+func main() {
+  // Use a GET request to look up the latest data on Charizard
+	resp, _ := http.Get("https://pokeapi.co/api/v2/pokemon/charizard")
+	body, _ := ioutil.ReadAll(resp.Body)
+
+  // Unmarshal the JSON into a struct
+	var data map[string]interface{}
+	json.Unmarshal(body, &data)
+
+  // Expose the pokemon data downstream to others steps in the "pokemon" key from this step
+	pd.Export("pokemon", data)
+}
+```
+
+Now this `pokemon` data is accessible to downstream steps within `pd.Steps.code.pokemon`
```

Comment:
Technically this won't work since Go does indexing in maps like `pd.Steps["code"]`